supermarq/feature_plotting.py

```diff
@@ -0,0 +1,135 @@
+from typing import Any, List, Optional, Tuple, Union
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.patches import Circle
+from matplotlib.projections import register_projection
+from matplotlib.projections.polar import PolarAxes
+
+
+def plot_benchmark(
+    data: List[Union[str, List[str], List[List[float]]]],
+    show: bool = True,
+    savefn: Optional[str] = None,
+    spoke_labels: Optional[List[str]] = None,
+    legend_loc: Tuple[float, float] = (0.75, 0.85),
+) -> None:
+    """
+    Create a radar plot of the given benchmarks.
+
+    Input
+    -----
+    data : Contains the title, feature data, and labels in the format:
+        [title, [benchmark labels], [[features_1], [features_2], ...]]
+    """
+    plt.rcParams["font.family"] = "Times New Roman"
+
+    if spoke_labels is None:
+        spoke_labels = ["Connectivity", "Liveness", "Parallelism", "Measurement", "Entanglement"]
+
+    N = len(spoke_labels)
+    theta = radar_factory(N, frame="circle")
+
+    _, ax = plt.subplots(dpi=150, subplot_kw=dict(projection="radar"))
+    # fig.subplots_adjust(wspace=0.25, hspace=0.20, top=0.85, bottom=0.05)
+
+    _, labels, case_data = data
+    # ax.set_rgrids([0.2, 0.4, 0.6, 0.8])
+    ax.set_rgrids([0.0, 0.2, 0.4, 0.6, 0.8, 1.0])
+    # ax.set_title(title, weight='bold', size='medium', position=(0.5, 1.1),
+    #             horizontalalignment='center', verticalalignment='center')
+    for d, label in zip(case_data, labels):
+        ax.plot(theta, d, label=label)
+        ax.fill(theta, d, alpha=0.25)
+    ax.set_varlabels(spoke_labels)
+
+    ax.legend(loc=legend_loc, labelspacing=0.1, fontsize=11)
+    plt.tight_layout()
+
+    if savefn is not None:
+        plt.savefig(savefn)  # pragma: no cover
+
+    if show:
+        plt.show()  # pragma: no cover
+
+    plt.close()
+
+
+def radar_factory(num_vars: int, frame: str = "circle") -> np.ndarray:
+    """
+    (https://matplotlib.org/stable/gallery/specialty_plots/radar_chart.html)
```

Comment:
Change docstring to be in the style of Google's recommendation: https://google.github.io/styleguide/pyguide.html